panel/lxqtpanelapplication.h

```diff
@@ -39,37 +39,140 @@ namespace LXQt {
 class Settings;
 }
 
+/*!
+ * \brief The LXQtPanelApplication class inherits from LXQt::Application and
+ * is therefore the QApplication that we will create and execute in our
+ * main()-function.
+ *
+ * LXQtPanelApplication itself is not a visible panel, rather it is only
+ * the container which holds the visible panels. These visible panels are
+ * LXQtPanel objects which are stored in mPanels. This approach enables us
+ * to have more than one panel (for example one panel at the top and one
+ * panel at the bottom of the screen) without additional effort.
+ */
 class LXQtPanelApplication : public LXQt::Application
 {
     Q_OBJECT
 public:
+    /*!
+     * \brief Creates a new LXQtPanelApplication with the given command line
+     * arguments. Performs the following steps:
+     * 1. Initializes the LXQt::Application, sets application name and version.
+     * 2. Handles command line arguments. Currently, the only cmdline argument
+     * is -c = -config = -configfile which chooses a different config file
+     * for the LXQt::Settings.
+     * 3. Creates the LXQt::Settings.
+     * 4. Connects QCoreApplication::aboutToQuit to cleanup().
+     * 5. Calls addPanel() for each panel found in the config file. If there is
+     * none, adds a new panel.
+     * \param argc
+     * \param argv
+     */
     explicit LXQtPanelApplication(int& argc, char** argv);
     ~LXQtPanelApplication();
 
+    /*!
+     * \brief Determines the number of LXQtPanel objects
+     * \return the current number of LXQtPanel objects
+     */
     int count() { return mPanels.count(); }
+    
+    /*!
+     * \brief Checks if a given Plugin is running and has the
+     * ILXQtPanelPlugin::SingleInstance flag set. As Plugins are added to
+     * LXQtPanel instances, this method only iterates over these LXQtPanel
+     * instances and lets them check the conditions.
+     * \param pluginId TODO
```

Comment:
This is the plugin identifier -> basename of .desktop file w/o the .desktop extension: e.g.  clock.desktop -> clock